CDP4Reporting/ViewModels/ReportDesignerViewModel.cs

```diff
@@ -289,39 +344,42 @@ public ReportDesignerViewModel(Iteration thing, ISession session, IThingDialogNa
             this.DataSourceTextChangedCommand = ReactiveCommand.Create();
             this.DataSourceTextChangedCommand.Subscribe(_ => this.CheckAutoCompileScript());
 
-            this.RebuildDatasourceCommand = ReactiveCommand.CreateAsyncTask(async _ =>
-            {
-                this.IsBusy = true;
-
-                try
-                {
-                    var source = this.Document.Text;
-                    await this.compilationConcurrentActionRunner.RunAction(() => this.CompileAssembly(source));
+            this.RebuildDatasourceCommand = ReactiveCommand.CreateAsyncTask(async _ => await this.ExecuteRebuildDatasourceCommand());
 
-                    if (!this.CompileResult?.Errors.HasErrors ?? false)
-                    {
-                            this.RebuildDataSource();
-                    }
-                }
-                finally
-                {
-                    this.IsBusy = false;
-                }
-            });
+            this.SubmitParameterValuesCommand = ReactiveCommand.CreateAsyncTask(
+                this.WhenAnyValue(x => x.CanSubmitParameterValues),
+                x => this.SubmitParameterValues(),
+                RxApp.MainThreadScheduler);
 
             this.ClearOutputCommand = ReactiveCommand.Create();
             this.ClearOutputCommand.Subscribe(_ => { this.Output = string.Empty; });
 
             this.ActiveDocumentChangedCommand = ReactiveCommand.CreateAsyncTask(x =>
                 this.SetReportDesigner(((DependencyPropertyChangedEventArgs) x).NewValue), RxApp.MainThreadScheduler);
 
+            this.WhenAnyValue(x => x.CurrentReport).Subscribe(x =>
+            {
+                x.AfterPrint += this.CheckSubmittableParameterValues;
+            });
+
+            this.Changing
+                .Where(x => x.PropertyName == nameof(this.CurrentReport))
+                .Subscribe(x => 
+                {
+                    if (this.CurrentReport != null)
+                    {
+                        this.CurrentReport.AfterPrint -= this.CheckSubmittableParameterValues;
+                    }
+                });
+
             this.InitializeDataSetExtensionsUsage();
         }
 
         /// <summary>
         /// Method that is here that does nothing, but makes sure that System.Data.DataSetExtensions.dll is
         /// available in the report script
         /// </summary>
+        [SuppressMessage("Minor Code Smell", "S1481:Unused local variables should be removed", Justification = "<Pending>")]
```

Comment:
This should be linked to an issue.

Reply:
No, because it is not an issue. Th summary tells why this is here. 
Changed the justification though.